### PR TITLE
util/perl/OpenSSL/Ordinals.pm: introduce a base version

### PR DESCRIPTION
**Pull request by levitte**
The idea is that a base version is the minimum version that must be
assigned to all symbols.  The practical result is that, for any new
major release, the version number for all symbols will automatically
be bumped to the new release's version number, if necessary.

(this also includes a few small fixup commits)

### CODE REVIEW

**Comment by levitte** [comment]
Ping?

**Review by t-j-h — APPROVED** [review]

File: util/perl/OpenSSL/Util.pm [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+=head1 DESCRIPTION
+
+=over
+
+=item B<cmp_versions "VERSION1", "VERSION2">
+
+Compares VERSION1 with VERSION2, paying attention to OpenSSL versioning.
+
+Returns 1 if VERSION2 is greater than VERSION1, 0 if they are equal, and
+-1 if VERSION1 is greater than VERSION2.
```
Comment (mattcaswell) [inline util/perl/OpenSSL/Util.pm]: Isn't this the other way around, i.e. -1 if VERSION1 is less than VERSION2 and 1 if VERSION1 is greater than VERSION2?

Comment (levitte) [reply util/perl/OpenSSL/Util.pm]: Oh, you're right! Will fix

Comment (levitte) [reply util/perl/OpenSSL/Util.pm]: It was doing the right thing in practice :laughing:

Comment (levitte) [reply util/perl/OpenSSL/Util.pm]: Fixup pushed

**Review by mattcaswell — APPROVED** [review]
Approved assuming the typo is fixed.

File: util/perl/OpenSSL/Util.pm [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+=head1 DESCRIPTION
+
+=over
+
+=item B<cmp_versions "VERSION1", "VERSION2">
+
+Compares VERSION1 with VERSION2, paying attention to OpenSSL versioning.
+
+Returns 1 if VERSION1 is greater than VERSION2, 0 if they are equal, and
+-1 if VERSION1 is lesser than VERSION2.
```
Comment (mattcaswell) [inline util/perl/OpenSSL/Util.pm]: "less than" not "lesser than"

Comment (levitte) [reply util/perl/OpenSSL/Util.pm]: So it's "greater than" but "less than"...

... English, the language where logic said buh-bye :wink: ...

Will fix (approval noted)

**Comment by levitte** [comment]
Merged.  Thanks for the reviews!

d26d2cfe955db15f7137247af50179dc94262e9d Remove all NOEXIST symbols from ordinals files
3b39185880a996787daf67bb74cb2a89e91e38cf make update
bfc3b4ffd6fab946c34e567d0afa8ea132e01f24 util/perl/OpenSSL/Ordinals.pm: introduce a base version
ad0b144b8a15c4d016bdef578c0ebfc72970da2c util/perl/OpenSSL/Ordinals.pm: use OpenSSL::Util::cmp_versions
257ab867d0f290fcfe5e1abe005086b8bd94d047 util/perl/OpenSSL/Ordinals.pm: shift to dotted versions internally

**Pull request closed without merging**